### PR TITLE
feat(competitions): add a page for autoware related competitions

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Description
Create a page for autoware related competitions

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/.pages [hunk 1]
```diff
@@ -7,3 +7,4 @@ nav:
   - contributing
   - datasets
   - support
+  - Competitions: autoware-competitions/index.md
```
Comment (yuasabe) [inline docs/.pages:10]: It looked like when the `index.md` file is specified, the link in the nav bar is not working: https://autowarefoundation.github.io/autoware-documentation/pr-382/autoware-competitions/index.md

Comment (yuasabe) [reply docs/.pages]: resolved in https://github.com/autowarefoundation/autoware-documentation/pull/382/commits/cef9e8f26303bbde4d0c84127d8a3460f59de90a !

**Comment by mitsudome-r** [comment]
Rebased the branch to signoff commits.

File: docs/index.md [hunk 2]
```diff
@@ -21,3 +21,4 @@ For detailed documents of Autoware Universe components, see [Autoware Universe D
 - [Contributing](contributing) pages explain how to contribute to Autoware.
 - [Datasets](datasets) pages contain information about datasets that can be used with Autoware.
 - [Support](support) pages explain several support resources.
+- [Competitions](autoware-competitions) pages explain on-going challenges releated to AWF
```
Comment (taikitanaka3) [inline docs/index.md:24]:
```suggestion
- [Competitions](autoware-competitions) pages explain on-going challenges related to AWF
```

**Pull request merged**